src/sentry/static/sentry/app/views/settings/components/dataScrubbing/modals/modalManager.tsx

```diff
@@ -0,0 +1,300 @@
+    const {eventId, sourceSuggestions} = fetchSourceGroupData();
+    const values = this.getInitialValues();
+    return {
+      values,
+      requiredValues: this.getRequiredValues(values),
+      errors: {},
+      isFormValid: false,
+      eventId: {
+        value: eventId,
+        status: !eventId ? EventIdStatus.UNDEFINED : EventIdStatus.LOADED,
+      },
+      sourceSuggestions,
+    } as Readonly<State>;
+  }
+
+  getInitialValues() {
+    const {initialState} = this.props;
+    return {
+      type: initialState?.type ?? RuleType.CREDITCARD,
+      method: initialState?.method ?? MethodType.MASK,
+      source: initialState?.source ?? '',
+      placeholder: initialState?.placeholder ?? '',
+      pattern: initialState?.pattern ?? '',
+    };
+  }
+
+  getRequiredValues(values: Values) {
+    const {type} = values;
+    const requiredValues: Array<KeysOfUnion<Values>> = ['type', 'method', 'source'];
+
+    if (type === RuleType.PATTERN) {
+      requiredValues.push('pattern');
+    }
+
+    return requiredValues;
+  }
+
+  clearError<F extends keyof Values>(field: F) {
+    this.setState(prevState => ({
+      errors: omit(prevState.errors, field),
+    }));
+  }
+
+  async loadSourceSuggestions() {
+    const {orgSlug, projectId, api} = this.props;
+    const {eventId} = this.state;
+
+    if (!eventId.value) {
+      this.setState(prevState => ({
+        sourceSuggestions: valueSuggestions,
+        eventId: {
+          ...prevState.eventId,
+          status: EventIdStatus.UNDEFINED,
+        },
+      }));
+      return;
+    }
+
+    this.setState(prevState => ({
+      sourceSuggestions: valueSuggestions,
+      eventId: {
+        ...prevState.eventId,
+        status: EventIdStatus.LOADING,
+      },
+    }));
+
+    try {
+      const query: {projectId?: string; eventId: string} = {eventId: eventId.value};
+      if (projectId) {
+        query.projectId = projectId;
+      }
+      const rawSuggestions = await api.requestPromise(
+        `/organizations/${orgSlug}/data-scrubbing-selector-suggestions/`,
+        {query}
+      );
+      const sourceSuggestions: SourceSuggestions = rawSuggestions.suggestions;
+
+      if (sourceSuggestions && sourceSuggestions.length > 0) {
+        this.setState(prevState => ({
+          sourceSuggestions,
+          eventId: {
+            ...prevState.eventId,
+            status: EventIdStatus.LOADED,
+          },
+        }));
+        return;
+      }
+
+      this.setState(prevState => ({
+        sourceSuggestions: valueSuggestions,
+        eventId: {
+          ...prevState.eventId,
+          status: EventIdStatus.NOT_FOUND,
+        },
+      }));
+    } catch {
+      this.setState(prevState => ({
+        eventId: {
+          ...prevState.eventId,
+          status: EventIdStatus.ERROR,
+        },
+      }));
+    }
+  }
+
+  convertRequestError(error: ReturnType<typeof handleError>) {
+    switch (error.type) {
+      case 'invalid-selector':
```

Comment:
Should these keys be in an enum?

Reply:
I've seen we using both. Please why do you think Enum would be better in this case?